bootstrap.sh

```diff
@@ -1,117 +1,308 @@
+		echo "ssh_public_key: \"${ssh_key_pair}\"" |tee -a "${CUSTOM_FILE}"
+	}
+}
+
+
+# Main
+
+ANSIBLE_WORK_DIR="${HOME}/ansible-easy-vpn"
+GITHUB_REPO="https://github.com/notthebee/ansible-easy-vpn"
+
+#### WHY? It kills the script with this heere... ?????
+## Discard stdin. Needed when running from an one-liner which includes a newline
+#read -r -N 999999 -t 0.001
+
+## Quit on error
+## -- already set in shebang line ????
+#set -e
+
+check_os
+install_packages_for_ansible_and_dependencies
+install_and_setup_homedir_files
+
+
+# Set secure permissions for the Vault file
+SECRET_FILE="${HOME}/ansible-easy-vpn/secret.yml"
+[[ -f "${SECRET_FILE}" ]] && {
+	echo
+	echo "WARNING:"
+	echo "${SECRET_FILE} already exists..."
+	echo
+}
+chmod 600 "${SECRET_FILE}"
+
+
+# Permissions are not critical with the CUSTOM_FILE
+# - secrets are not kept in this file
+CUSTOM_FILE="${HOME}/ansible-easy-vpn/custom.yml"
+[[ -f "${CUSTOM_FILE}" ]] && {
+	echo
+	echo "WARNING:"
+	echo "${CUSTOM_FILE} already exists..."
+	echo
+}
 
-# Check if we're running on an AWS EC2 instance
-set +e
-aws=$(curl -m 5 -s http://169.254.169.254/latest/meta-data/ami-id)
 
-if [[ "$aws" =~ ^ami.*$ ]]; then
-  aws=true
-else
-  aws=false
-fi
-set -e
-touch $HOME/ansible-easy-vpn/custom.yml
 
 
 
 clear
 echo "Welcome to ansible-easy-vpn!"
 echo
 echo "This script is interactive"
-echo "If you prefer to fill in the custom.yml file manually,"
+echo "If you prefer to fill in the ${CUSTOM_FILE} file manually,"
 echo "press [Ctrl+C] to quit this script"
 echo
 echo "Enter your desired UNIX username"
-read -p "Username: " username
-until [[ "$username" =~ ^[a-z0-9]*$ ]]; do
-  echo "Invalid username"
-  echo "Make sure the username only contains lowercase letters and numbers"
-  read -p "Username: " username
+username=
+until [[ ${username} =~ ^[a-z0-9]*$ && -n ${username} ]]; do
+	[[ -n ${username} ]] && echo "Invalid username"
+	echo "Make sure the username only contains lowercase letters and numbers"
+	read -r -p "Username: " username
 done
+# First write to CUSTOM_FILE, old content will be lost
+echo "username: \"${username}\"" |tee "${CUSTOM_FILE}"
 
-echo "username: \"${username}\"" >> $HOME/ansible-easy-vpn/custom.yml
 
 echo
 echo "Enter your user password"
 echo "This password will be used for Authelia login, administrative access and SSH login"
-read -s -p "Password: " user_password
-until [[ "${#user_password}" -lt 60 ]]; do
-  echo
-  echo "The password is too long"
-  echo "OpenSSH does not support passwords longer than 72 characters"
-  read -s -p "Password: " user_password
-done
-echo
-read -s -p "Repeat password: " user_password2
-echo
-until [[ "$user_password" == "$user_password2" ]]; do
-  echo
-  echo "The passwords don't match"
-  read -s -p "Password: " user_password
-  echo
-  read -s -p "Repeat password: " user_password2
+echo "Passwords longer trhan 72 bytes are not supported by OpenSSH private key format"
+echo " -- is this really true these days???"
```

Comment:
Aparently – #4

Reply:
Not sure if needed – in the unlikely case where the user does have a 72+ symbol password, he'll be warned that it's too long anyway